### PR TITLE
 Initializing the restic repository 

### LINKED ISSUE #16: Repository initialization
This role lack repository initialization, which is kind of crucial in new production systems.

### PR DESCRIPTION
**Pull request by donat-b**
Resolves #16

### CODE REVIEW

File: tasks/configure.yml [hunk 1]
```diff
@@ -7,3 +7,27 @@
...
+  command: '{{ restic_install_path }}/restic snapshots'
+  environment:
+    RESTIC_REPOSITORY: '{{ item.url }}'
+    RESTIC_PASSWORD: '{{ item.password }}'
+  register: _repo_check
+  ignore_errors: True
+  changed_when: False
+  with_items: '{{ restic_repos }}'
+  when:
+    - item.init is defined
```
Comment (paulfantom) [inline tasks/configure.yml]: Define it as `False` in `defaults/main.yml` and don't check if it is defined. That's what default values are for.

File: tasks/configure.yml [hunk 2]
```diff
@@ -7,3 +7,27 @@
     mode: '0640'
   no_log: true
   with_items: '{{ restic_repos }}'
+
+- name: Check if the repo already exists
```
Comment (paulfantom) [inline tasks/configure.yml]: If you are checkint if repository exists what is the purpose of `init` variable?

Comment (donat-b) [reply tasks/configure.yml]: I'm not completely sure in validity of this kind of check and think it's better not to take any actions on repository unless asked to.

Comment (paulfantom) [reply tasks/configure.yml]: Did you check what happens if you initialize same repository twice? If initialization (`restic init`) is idempotent then this kid of check is useless.

Comment (donat-b) [reply tasks/configure.yml]: Just checked and it seems like it is. But removing the check would mean I'll have to ignore errors (it returns a non-zero code) and you won't see an actual error. So the check should stay.

Comment (paulfantom) [reply tasks/configure.yml]: That's what `failed_when` option is for. To decide what to fail based not on exit code but on messages it returns.

Comment (paulfantom) [reply tasks/configure.yml]: Here is how it works: http://docs.ansible.com/ansible/latest/playbooks_error_handling.html#controlling-what-defines-failure

Comment (donat-b) [reply tasks/configure.yml]: Great! I'll look into that

**Comment by donat-b** [comment]
@paulfantom merging this?

**Comment by paulfantom** [comment]
Looks good, I would add `no_log` to `command` since we are passing password in `environment` section.

**Comment by donat-b** [comment]
Good thought.

**Pull request merged**